packages/vats/src/vat-bank.js

```diff
@@ -0,0 +1,269 @@
+
+      /**
+       * @typedef {Object} AssetDescriptor
+       * @property {Brand} brand
+       * @property {Issuer} issuer
+       * @property {string} issuerName
+       * @property {string} denom
+       * @property {string} proposedName
+       */
+      /** @type {SubscriptionRecord<AssetDescriptor>} */
+      const {
+        subscription: assetSubscription,
+        publication: assetPublication,
+      } = makeSubscriptionKit();
+
+      /** @type {Store<string, Bank>} */
+      const addressToBank = makeStore('address');
+      return Far('bankManager', {
+        /**
+         * Returns assets as they are added to the bank.
+         *
+         * @returns {Subscription<AssetDescriptor>}
+         */
+        getAssetSubscription() {
+          return harden(assetSubscription);
+        },
+        /**
+         * Add an asset to the bank, and publish it to the subscriptions.
+         *
+         * @param {string} denom lower-level denomination string
+         * @param {string} issuerName
+         * @param {string} proposedName
+         * @param {AssetIssuerKit} kit ERTP issuer kit (mint, brand, issuer)
+         */
+        async addAsset(denom, issuerName, proposedName, kit) {
+          assert.typeof(denom, 'string');
+          assert.typeof(issuerName, 'string');
+          assert.typeof(proposedName, 'string');
+
+          const brand = await kit.brand;
+          const mathKind = await E(kit.issuer).getAmountMathKind();
+          assert.equal(
+            mathKind,
+            MathKind.NAT,
+            `Only fungible assets are allowed, not ${mathKind}`,
+          );
+
+          const assetRecord = harden({ ...kit, denom, brand });
+          brandToAssetRecord.init(brand, assetRecord);
+          denomToAddressUpdater.init(denom, makeStore('address'));
+          assetPublication.updateState(
+            harden({
+              brand,
+              denom,
+              issuerName,
+              issuer: kit.issuer,
+              proposedName,
+            }),
+          );
+        },
+        /**
+         * Create a new personal bank interface for a given address.
+         *
+         * @param {string} address lower-level bank account address
+         * @returns {Bank}
+         */
+        getBankForAddress(address) {
+          assert.typeof(address, 'string');
+          if (addressToBank.has(address)) {
+            return addressToBank.get(address);
+          }
+
+          /** @type {Store<Brand, VirtualPurse>} */
+          const brandToVPurse = makeStore('brand');
+
+          /** @type {Bank} */
+          const bank = Far('bank', {
+            getAssetSubscription() {
+              return harden(assetSubscription);
+            },
+            async getPurse(brand) {
+              if (brandToVPurse.has(brand)) {
+                return brandToVPurse.get(brand);
+              }
+
+              const assetRecord = brandToAssetRecord.get(brand);
+              if (!bankCall) {
```

Comment:
when would this case be triggered? If bankCall is false, don't we want to just error? Or is this the simchain? Can we add a comment explaining?

Reply:
It's the sim-chain.  I will clarify.